solver/pb/ops.proto

```diff
@@ -60,8 +62,19 @@ message BuildOp {
 	repeated bytes def = 3;
 	map<string, string> attrs = 4;
 	// outputs
+	// TODO: add Metadata (bytes?) for def
 }
 
 message BuildInput {
 	int64 input = 1 [(gogoproto.customtype) = "InputIndex", (gogoproto.nullable) = false];
+}
+
+message Metadata {
```

Comment:
Is this really related to `LLB` definition at all, or controller definition? How about defining 
```
struct {
   Definition [][]byte
   Metadata map[digest.Digest]MetadataEntry
}
```
 that would be what `llb.Marshal` generates and `client.Solve` takes as an input. `buildctl` can also understand that and override if needed. Then a script generating `llb` can also define it. For example, worker constraints need to be defined by the script not by the override in `buildctl`.